### PR TITLE
Added form for add customer and format code

### PR DESCRIPTION
**Pull request by younessoudouh**

### CODE REVIEW

File: src/components/FilterPopUp/FilterPopUp.jsx [hunk 1]
```diff
@@ -1,95 +1,95 @@
...
+      <div className="sort-by-name">
+        <h3 className="sort-title">sort by:</h3>
+        <Input
+          htmlFor="sort-default"
+          type="radio"
+          id="sort-default"
+          value="sort-default"
+          changeHandler={handleSortByNameChange}
+          checked={sort.name === "sort-default"}
+          text="Default"
```
Comment (faouzioudouh) [inline src/components/FilterPopUp/FilterPopUp.jsx]: What is this?

File: src/components/Form/Form.jsx [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+  const inputRef = useRef();
+
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+    let isFormValid = Object.values(errors).every(
+      (error) => error.length === 0
```
Comment (faouzioudouh) [inline src/components/Form/Form.jsx]: What do you mean by `error.length === 0` ?

File: src/components/Form/Form.jsx [hunk 3]
```diff
@@ -297,71 +312,125 @@ const Form = ({
           refer={inputRef}
         />
         <FormInput
+          className={
+            !customerInfo.lastName && !errors.lastName
+              ? ""
+              : errors.lastName
+              ? "error"
+              : "success"
```
Comment (faouzioudouh) [inline src/components/Form/Form.jsx]: Can you simplify this condition a bit, its hard to read?

File: src/components/Form/Form.jsx [hunk 4]
```diff
@@ -286,6 +294,13 @@ const Form = ({
       <h2 className="form-header">Add Customer</h2>
       <section className="first-section">
         <FormInput
+          className={
+            !customerInfo.firstName && !errors.firstName
+              ? ""
+              : errors.firstName
+              ? "error"
+              : "success"
```
Comment (faouzioudouh) [inline src/components/Form/Form.jsx]: Can you simplify this condition a bit, its hard to read?

File: src/components/Form/Form.jsx [hunk 5]
```diff
@@ -32,26 +31,35 @@ const Form = ({
...
+        inputsErrors = {
+          ...inputsErrors,
           [inputName]: inputName + " can`t be blank",
-        }))
+        };
+      }
+    });
+
+    let isFormValid = Object.values(inputsErrors).every(
+      (error) => error.length === 0
```
Comment (faouzioudouh) [inline src/components/Form/Form.jsx]: What does it mean `error.length === 0`?

File: src/components/Form/Form.jsx [hunk 6]
```diff
@@ -0,0 +1,474 @@
+import React, { useEffect, useRef, useState } from "react";
+import "./Form.css";
+import FormInput from "../FormInput/FormInput";
+import Button from "../Button/Button";
+import InputError from "../InputError/InputError";
+import FormSelect from "../FormSelect/FormSelect";
+
+const Form = ({
```
Comment (faouzioudouh) [inline src/components/Form/Form.jsx:8]: This component is big and holds a lot of logic, can we try to break it down?

File: src/components/FormInput/FormInput.jsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        type={type}
+        name={name}
+        placeholder={placeholder}
+        maxLength={maxLength}
+        className={`add-customer-input ${className}`}
+        value={value}
+        ref={refer}
+      />
+      {!isInputValid ? <InputError errorMessage={errorMessage} /> : null}
+      {isInputValid ? <i className="fas fa-check-circle"></i> : null}
```
Comment (faouzioudouh) [inline src/components/FormInput/FormInput.jsx]:
```suggestion
      {!isInputValid ? <InputError errorMessage={errorMessage} /> : <i className="fas fa-check-circle"></i>}
```

File: src/components/Modal/Modal.css [hunk 8]
```diff
@@ -0,0 +1,13 @@
+.modal {
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+    /* backdrop-filter: blur(1px); */
```
Comment (faouzioudouh) [inline src/components/Modal/Modal.css:7]: Remove unused code

File: src/components/Form/Form.jsx (hunk 5 above)
Comment (younessoudouh) [reply src/components/Form/Form.jsx]: to check inputs errors if input error is empty that's mean input is valid

File: src/components/FilterPopUp/FilterPopUp.jsx (hunk 1 above)
Comment (younessoudouh) [reply src/components/FilterPopUp/FilterPopUp.jsx]: its make the sort input checkbox checked depend on the state

File: src/components/Form/Form.jsx (hunk 2 above)
Comment (younessoudouh) [reply src/components/Form/Form.jsx]: input error is empty

File: src/components/Form/Form.jsx (hunk 6 above)
Comment (younessoudouh) [reply src/components/Form/Form.jsx:8]: could you give me a hint ?

File: src/components/Form/Form.jsx [hunk 9]
```diff
@@ -22,65 +26,114 @@ const Form = ({
...
   useEffect(() => {
+    if (editOpen) {
+      setCustomerInfo(customerToEdit);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (firstRender.current) {
+      return;
+    } else if (firstRender.current) {
```
Comment (younessoudouh) [inline src/components/Form/Form.jsx]: duplicted I corrected it already

**Pull request merged**